src/methods.jl

```diff
@@ -1,102 +1,147 @@
+    perform adaptation for this finite difference method.
+- `condition::Float64`: Condition number. See See [`DEFAULT_CONDITION`](@ref).
+- `factor::Float64`: Factor number. See See [`DEFAULT_FACTOR`](@ref).
+- `max_range::Float64`: Maximum distance that a function is evaluated from the input at
+    which the derivative is estimated.
+- `∇f_magnitude_mult::Float64`: Internally computed quantity.
+- `f_error_mult::Float64`: Internally computed quantity.
 """
-const DEFAULT_CONDITION = 100
+struct UnadaptedFiniteDifferenceMethod{P,Q} <: FiniteDifferenceMethod{P,Q}
+    grid::SVector{P,Int}
+    coefs::SVector{P,Float64}
+    coefs_neighbourhood::NTuple{3,SVector{P,Float64}}
+    condition::Float64
+    factor::Float64
+    max_range::Float64
+    ∇f_magnitude_mult::Float64
+    f_error_mult::Float64
+end
 
 """
-    FiniteDifferenceMethod{G<:AbstractVector, C<:AbstractVector, E<:Function}
+    AdaptedFiniteDifferenceMethod{
+        P,
+        Q,
+        E<:FiniteDifferenceMethod
+    } <: FiniteDifferenceMethod{P,Q}
 
-A finite difference method.
+A finite difference method that estimates a `Q`th order derivative from `P` function
+evaluations. This method does dynamically adapt its step size.
 
 # Fields
-- `grid::G`: Multiples of the step size that the function will be evaluated at.
-- `q::Int`: Order of derivative to estimate.
-- `coefs::C`: Coefficients corresponding to the grid functions that the function evaluations
-    will be weighted by.
-- `bound_estimator::Function`: A function that takes in the function and the evaluation
-    point and returns a bound on the magnitude of the `length(grid)`th derivative.
+- `grid::SVector{P,Int}`: Multiples of the step size that the function will be evaluated at.
+- `coefs::SVector{P,Float64}`: Coefficients corresponding to the grid functions that the
+    function evaluations will be weighted by.
+- `coefs_neighbourhood::NTuple{3,SVector{P,Float64}}`: Sets of coefficients used for
+    estimating the magnitude of the derivative in a neighbourhood.
+- `condition::Float64`: Condition number. See See [`DEFAULT_CONDITION`](@ref).
+- `factor::Float64`: Factor number. See See [`DEFAULT_FACTOR`](@ref).
+- `max_range::Float64`: Maximum distance that a function is evaluated from the input at
+    which the derivative is estimated.
+- `∇f_magnitude_mult::Float64`: Internally computed quantity.
+- `f_error_mult::Float64`: Internally computed quantity.
+- `bound_estimator::FiniteDifferenceMethod`: A finite difference method that is tuned to
+    perform adaptation for this finite difference method.
 """
-struct FiniteDifferenceMethod{G<:AbstractVector, C<:AbstractVector, E<:Function}
-    grid::G
-    q::Int
-    coefs::C
+struct AdaptedFiniteDifferenceMethod{
+    P,
+    Q,
+    E<:FiniteDifferenceMethod
+} <: FiniteDifferenceMethod{P,Q}
+    grid::SVector{P,Int}
+    coefs::SVector{P,Float64}
+    coefs_neighbourhood::NTuple{3,SVector{P,Float64}}
+    condition::Float64
```

Comment:
What's the reason for making these `Float64`s rather than making this more generic?

Reply:
Hmm, because the coefficients in `_coefs_mults` are also converted to `Float64`s. I guess we could add a type parameter to parametrise the operating data type for internal computations. Would you be in favour of that?

Reply:
I can't say that I'm overly fussed either way. Provided that you don't think it's a problem, we could just go with`Float64` until need to generalise.